src/services/getEditsForFileRename.ts

```diff
@@ -44,8 +44,11 @@ namespace ts {
                 // If it resolved to something already, ignore.
                 if (checker.getSymbolAtLocation(importStringLiteral)) continue;
 
-                const resolved = program.getResolvedModuleWithFailedLookupLocationsFromCache(importStringLiteral.text, sourceFile.fileName);
-                if (resolved && contains(resolved.failedLookupLocations, oldFilePath)) {
+                if (!host.resolveModuleNameWithFailedLookupLocations && !host.fileExists) continue;
+                const resolved = host.resolveModuleNameWithFailedLookupLocations
+                    ? host.resolveModuleNameWithFailedLookupLocations(importStringLiteral.text, sourceFile.fileName)
+                    : resolveModuleName(importStringLiteral.text, sourceFile.fileName, program.getCompilerOptions(), host as ModuleResolutionHost);

```

Comment:
This is not quite as i suggested, here you are still relying on ```resolveModuleName``` which you shouldnt . You should always respect host.resolveModuleName implmentation. That leaves us with only following option. You  **should not re-resolve module that too ignoring host's resolution method.** (re-resolving is also costly esp if you are renaming in project with 1000+ files)

1. when host.resolveModuleName is given
   Make host.resolveModuleNameWithFailedLookupLocations mandatory for us to get the failed look up locations. 
2. when its not provided => compiler uses resolveModuleName with the cache. I think you should use the cache here instead of re-resolving since fileExists and other just operations can be costly. 

Irrespective of the any of the above methods, these should operate on cache and should be "get" like instead of resolve since you want the existing result instead of getting new one.